--- .env/lib/python3.8/site-packages/geneticalgorithm/geneticalgorithm.py.back	2021-06-22 16:52:53.822289660 +0430
+++ .env/lib/python3.8/site-packages/geneticalgorithm/geneticalgorithm.py	2021-06-22 16:52:58.910257767 +0430
@@ -454,6 +454,8 @@
         if self.stop_mniwi==True:
             sys.stdout.write('\nWarning: GA is terminated due to the'+\
                              ' maximum number of iterations without improvement was met!')
+
+        return self.best_variable, self.best_function, self.report
 ##############################################################################         
 ##############################################################################         
     def cross(self,x,y,c_type):
